src/store/audio.ts

```diff
@@ -939,9 +1032,8 @@ export const audioStore = createPartialStore<AudioStoreTypes>({
     },
   },
 
-  APPLY_AUDIO_PRESET: {
-    mutation(state, { audioKey }: { audioKey: AudioKey }) {
-      const audioItem = state.audioItems[audioKey];
+  APPLY_AUDIO_PRESET_TO_AUDIO_ITEM: {
```

Comment:
`APPLY_AUDIO_PRESET` が登録済みでaudioKeyが発行されているものにしかapplyできなかったので、
野良のaudioItemに対してプリセットをapplyできるように分解しました
`GENERATE_AUDIO_ITEM` 内でプリセット適用するために必要でした

Reply:
> 引数に与えられたAudioItemを書き換えていて

既存コードなのでリファクタPRで直しますね

Reply:
既存コードは「keyが与えられ、stateからAudioItemを取得して書き換える」という感じで、ここで「AudioItemを与えられ、そのAudioItemを書き換える」になっているかなと思いました！

まあでもVuex側でそういうことしてはいけないと明確に書かれてるわけじゃないので必ず直さないといけないことはなさそう。
このPR内ではとりあえずFIXMEコメントを書く感じでどうでしょう 👀 
```suggestion
  APPLY_AUDIO_PRESET_TO_AUDIO_ITEM: {
    // FIXME: audioItemを直接書き換えないようにするか、関数化する
```

Reply:
なるほど理解しました
あ～～～そうするとaudioKeyを発行してからじゃないとstate内でaudioItemにアクセスできないんですね。
なんで今までREGISTER_AUDIO_ITEMの中でプリセット適用してたんだろうと思ってたけどそういうことか・・・

うーんこれはGENERATE_AUDIO_ITEM内でパラメータ適用する、という方針転換がここで支えられているので、
根深いことになるかもと思っていますが、リファクタPRでなんとかさせてください・・・ 🙇 🙏 🙇

Reply:
そうですね。。。。（僕も今気づきました 🙇‍♂️ 🙇‍♂️ 🙇‍♂️ ）

リファクタPR了解です、助かります！！
今のまま`APPLY_AUDIO_PRESET_TO_AUDIO_ITEM`を普通の関数にする感じでもとりあえず解決する気はします。
うーーーん･･･。どうすればきれいなのか･･･。 😇 😇

Reply:
他のCOMMANDから `APPLY_AUDIO_PRESET` を呼んでいて、そこからさらに `APPLY_AUDIO_PRESET_TO_AUDIO_ITEM` を呼んでいるので、
mutationに乗っけておかないとUNDOのときに困らないか・・・？と思いましたがmutation内からその関数を呼べばいいだけですね。

関数には切り出せますが、vuexのmutations以外からは呼び出すべきではない関数になってしまうのでう～～～ん・・・
stateに紐づくaudioItemを変更する、という副作用を排した形
（与えられたaudioItemとpresetItemから、presetを適用した新しいaudioItemを返す）
で関数に切り出したら綺麗かなと思いますが、
結局それを `state.audioItems[audioKey] = presetAppliedAudioItem` する必要がある訳でaudioKeyが発行されてからじゃないと変更できないという・・・う～～～～ん

なんも思いついてないですがリファクタPRで未来の自分がなんとかしてくれるらしいです！！！！

Reply:
良さそうです！！！

> vuexのmutations以外からは呼び出すべきではない関数になってしまうのでう～～～ん・・・

これもたしかに･･････。
exportしない＆コメントでmutation内以外から呼ぶことを想定していないのを書くとかでしょうか･･･。